### PR TITLE
lint: replace deprecated golang.org/x/lint/golint with golangci-lint

### PR DESCRIPTION
**Pull request by pohly**
golint is deprecated. golangci-lint is a more than adequate replacement. Because it combines different linters, a single invocation in the Makefile is enough and then produces consistent output for all of them.

It also gets enabled as a GitHub action. This will annotate pull requests.

The godox linter gets enabled as replacement for the manual "grep -i fixme".

For the sake of convenience, "make lint" installs golangci-lint. In contrast to golint before, this is done without polluting the go.mod file with the extra dependencies.

Fixes: #92

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,18 +24,16 @@ cover:
...
+# .github/workflows/golangci-lint.yml
 $(GOLINT):
-	go install golang.org/x/lint/golint
+	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.51.0
 
 .PHONY: lint
 lint: $(GOLINT)
-	@rm -rf lint.log
-	@echo "Checking formatting..."
-	@gofmt -d -s $(GO_FILES) 2>&1 | tee lint.log
```
Comment (liggitt) [inline Makefile:34]: should the gofmt / go vet / fixme checks remain?

Comment (pohly) [reply Makefile:34]: "govet" is enabled by default in golangci-lint. I explicitly enabled "godox" as replacement for the grep  command.

What I got wrong is "gofmt": I thought it was enabled by default., but it isn't. Will fix....

... and of course it helps to actually include the `.golangci.yml` in the commit :hot_face:

**Comment by pohly** [comment]
@prashantv: do you want me to rebase or shall I close this PR?  Entirely up to you, I'm fine with both.

**Comment by prashantv** [comment]
@pohly Thanks for the PR, since revive is a drop-in replacement that runs faster, I put that up in #97.

I think let's keep both open to see what direction others prefer -- if golangci-lint is preferred, then this PR can be rebased.

**Comment by abhinav** [comment]
Thanks, @pohly.
This can probably be closed now that #97 is merged.
(golangci-lint is cool, but it does a lot more than golint. If we ever switch to it, we'll want to drop staticcheck as well since golangci-lint bundles it in too.)

**Comment by pohly** [comment]
Please don't forget to do [a release](https://github.com/uber-go/goleak/issues/99) :smile:

**Pull request closed without merging**